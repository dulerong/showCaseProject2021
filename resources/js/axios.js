import axios from 'axios'
import Vue from 'vue'
// import { store } from './store'
// import router from './router'
// import { routePath } from '~js/router/constantRoutes'
// import { apiPath } from './constantsAPI'

const baseURL = '/'

const myAPI = axios.create({
  baseURL: baseURL,
  timeout: 30000, // unit = millisecond
  headers: {
    "Content-Type": "application/json"
  }
})

// const getAuthToken = () => `Bearer ${localStorage.getItem('user-token')}`

// const authInterceptor = (config) => {
//   config.headers['Authorization'] = getAuthToken()
//   return config
// }

// myAPI.interceptors.request.use(authInterceptor)

// const errorInterceptor = err => {
//   switch(err.response.status) {
//     case 400:
//       console.log(err.response)
//       errorFourZeroZero(err)
//       break
//     case 401:
//       console.log(err.response)
//       errorFourZeroOne(err)
//     default:
//       console.log(err.response)
//       showNotification(err.response.status, '管理会社へお問合せください', 'error', -1)
//   }
//   store.commit('loading/loadingStop')
//   return Promise.reject(err)
// }

// const noAuthRedirect = () => {
//   localStorage.removeItem('user-token')
//   store.commit('auth/removeToken')
//   router.push(routePath.login.path)
// }

// const errorFourZeroZero = (err) => {
//   switch(err.response.config.url) {
//     case apiPath.customer.getDetail:
//       router.push(routePath.search.path)
//       showNotification('無効の顧客ID', '顧客検索ページに転送', 'error', -1)
//       break
//     case apiPath.auth.login:
//       showNotification('ログインIDかパスワードが間違っています', '', 'error', -1)
//       break
//     default:
//       showNotification(err.response.status, '管理会社へお問合せください', 'error', -1)
//   }
// }

// const errorFourZeroOne = (err) => {
//   switch(err.response.config.url) {
//     case apiPath.auth.changePassword:
//       break
//     default:
//       showNotification('ユーザー認証失敗', 'ログイン画面に転送', 'error', -1)
//       return noAuthRedirect()
//   }
// }

// const showNotification = (status, message, color, timeout) => {
//   store.commit('notification/showNotification', {
//     message: `エラー: ${status}, ${message}`,
//     color: color,
//     timeout: timeout
//   })
// }

// const responseInterceptor = res => {
//   switch(res.status) {
//     case 200:
//       console.log('response 200')
//       break;
//     default:
//       console.log(res)
//   }
//   return res
// }

// myAPI.interceptors.response.use(responseInterceptor, errorInterceptor)

export default myAPI

Vue.prototype.$_api = myAPI