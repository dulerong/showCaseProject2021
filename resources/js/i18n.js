import Vue from 'vue'
import VueI18n from 'vue-i18n'
import { language } from './plugins/language'

Vue.use(VueI18n)

const messages = language

export default new VueI18n({
  locale: 'en',
  messages
})