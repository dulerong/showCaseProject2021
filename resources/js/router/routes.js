// route paths
import { routePath } from './constantRoutes'

// layouts
import DefaultLayout from '~layouts/default'
import LoginLayout from '~layouts/login'

// pages
import HomePage from '~pages/HomePage'
import LoginPage from '~pages/LoginPage'
import StatPage from '~pages/StatPage'
import NotFoundPage from '~pages/NotFoundPage'

// import { store } from './store'

export default [
  {
    path: routePath.home.path,
    name: routePath.home.name,
    component: HomePage,
    meta: { layout: DefaultLayout }
  },
  {
    path: routePath.stat.path,
    name: routePath.stat.name,
    component: StatPage,
    meta: { layout: DefaultLayout }
  },
  {
    path: routePath.login.path,
    name: routePath.login.name,
    component: LoginPage,
    meta: { layout: LoginLayout }
  },
  {
    path: '*',
    component: NotFoundPage,
    meta: { layout: DefaultLayout, requiresAuth: true }
  }
]