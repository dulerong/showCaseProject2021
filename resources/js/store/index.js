import Vue from 'vue'
import Vuex from 'vuex'

// import { auth } from './storeModules/auth'
// import { customerManage } from './storeModules/customerManage'
// import { customerSearch } from './storeModules/customerSearch'
// import { customerInquiry } from './storeModules/customerInquiry'
// import { customerInsideSales } from './storeModules/customerInsideSales'
// import { customerUndecided } from './storeModules/customerUndecided'
// import { locationSearch } from './storeModules/locationSearch'
import { home } from './storeModules/home'
// import { loading } from './storeModules/loading'
// import { salesTheme } from './storeModules/salesTheme'
// import { notification } from './storeModules/notification'
// import { stat } from './storeModules/stat'
// import { calendar } from './storeModules/calendar'

Vue.use(Vuex)

export const store = new Vuex.Store({
  modules: {
    // auth,
    // customerManage,
    // customerSearch,
    // customerInquiry,
    // customerInsideSales,
    // customerUndecided,
    // locationSearch,
    home,
    // loading,
    // salesTheme,
    // notification,
    // stat,
    // calendar,
  }
})