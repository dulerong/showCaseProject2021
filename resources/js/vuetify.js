import Vue from 'vue'
import Vuetify from 'vuetify'
import 'vuetify/dist/vuetify.min.css'

// Translation provided by Vuetify (javascript)
import ch from 'vuetify/es5/locale/zh-Hant'
import en from 'vuetify/es5/locale/en'

Vue.use(Vuetify)

const opts = {
  lang: {
    locales: { en, ch },
    current: 'en'
  },
  theme: {
    themes: {
      light: {
        primary: '#1976D2',
        deleteBtn: '#CD5542',
        editBtn: '#E79F3C',
        detailBtn: '#F4F4F4',
        field: '#ECEFF1',
      }
    }
  }
}

export default new Vuetify(opts)