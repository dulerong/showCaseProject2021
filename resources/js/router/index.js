import Vue from 'vue'
import VueRouter from 'vue-router'
import routes from './routes'
import i18n from '../i18n'
import vuetify from '../vuetify'
// import { store } from './store'
// import { routePath } from './constantsRoutes'

Vue.use(VueRouter)

const router = new VueRouter({
  routes,
  mode: 'history',
  scrollBehavior() {
    return { x: 0, y: 0, behavior: 'smooth' }
  },
})

const langList = ['en', 'ch']

router.beforeEach((to, from, next) => {
  const { lang } = to.params
  if (!langList.includes(lang)) return next('en') // if requested language not available, set locale to english and redirect to homepage

  i18n.locale = lang // set current language for i18n.
  vuetify.framework.lang.current = lang // set current language for vuetify
  next()

  // if (to.path === '/') next({ name: 'login', params: { lang: 'en' } })
  // if (!langList.includes(lang)) next({ name: to.name, params: { lang: 'en' } })
  // else next()
})

// router.beforeEach((to, from, next) => {
//   if (to.meta.layout !== from.meta.layout) store.commit('notification/hideNotification')

//   if (to.matched.some(record => record.meta.requiresAuth)) {
//     store.getters['auth/isLoggedIn'] ? next() : next({ name: routePath.login.name })
//   } else if (to.matched.some(record => record.meta.requiresVisitor)) {
//     store.getters['auth/isLoggedIn'] ? next({ name: routePath.home.name }) : next()
//   } else {
//     next()
//   }
// })
  
export default router