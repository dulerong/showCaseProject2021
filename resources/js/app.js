// require('./bootstrap');

import Vue from 'vue'
import App from './vue/App.vue'
import vuetify from './vuetify'
import router from './router/'
import { store } from './store/'

// below are plugins
import './axios'
import './plugins/day'
import './plugins/validation'
import './plugins/statPageConstants'

Vue.config.productionTip = false

new Vue ({
  el: '#app',
  vuetify,
  router,
  store,
  components: { App }
})
