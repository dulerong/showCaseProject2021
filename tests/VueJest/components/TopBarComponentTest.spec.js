import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import routes from '~/js/routes'
import { routePath } from '~/js/constantsRoutes'

import TopBar from '~components/crm/TopBar'

describe('TopBar Component Test', () => {
  const localVue = createLocalVue()

  localVue.use(Vuex)
  localVue.use(VueRouter)

  let vuetify
  let actions
  let auth
  let home
  let homeState
  let homeActions
  let store
  let router

  const loggedUser = {
    id: "9324a608-62e5-43b3-829d-634119f92661",
    name: "Dr. Jaylon Parker",
    name_kana: "-",
    is_admin: 1,
    email: "admin@example.com",
    department_id: 2,
    department_name: "I-Style",
    branch_id: 4,
    branch_name: "周南店",
    company_id: 1,
    company_name: "イケダグループホールディングス"
  }

  beforeEach(() => {
    vuetify = new Vuetify()
    router = new VueRouter({ routes })
    actions = {
      authLogout: jest.fn()
    }
    homeState = {
      loggedUser
    },
    homeActions = {
      retrieveLoggedUser: jest.fn()
    }
    auth = {
      namespaced: true,
      actions
    }
    home = {
      namespaced: true,
      state: homeState,
      actions: homeActions
    }
    store = new Vuex.Store({
      modules: {
        auth,
        home,
      }
    })
  })

  const propsData = {
    toggle: jest.fn()
  }

  const mountFunction = options => {
    return mount(TopBar, {
      localVue,
      vuetify,
      store,
      router,
      propsData,
      ...options,
    })
  }

  it('Component renders properly', () => {
    const wrapper = mountFunction()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Profile account contains correct username', () => {
    const wrapper = mountFunction()

    const accountName = wrapper.find('.v-toolbar__title')
    expect(accountName.exists()).toBe(true)
    expect(accountName.text()).toBe(loggedUser.name)
  })

  it('Logout action fired when profile button clicked', async () => {
    const wrapper = mountFunction()

    const button = wrapper.findAll('.v-btn').at(1)
    expect(button.exists()).toBe(true)

    await button.trigger('click')
    await wrapper.vm.$nextTick()

    expect(actions.authLogout).toHaveBeenCalled()
    expect(wrapper.vm.$route.name).toBe(routePath.login.name)
  })

  it('Toggle is fired when nav bar icon is clicked', async () => {
    const wrapper = mountFunction()

    const button = wrapper.find('.v-app-bar__nav-icon')
    expect(button.exists()).toBe(true)

    await button.trigger('click')
    await wrapper.vm.$nextTick()

    expect(propsData.toggle).toHaveBeenCalled()
  })
})