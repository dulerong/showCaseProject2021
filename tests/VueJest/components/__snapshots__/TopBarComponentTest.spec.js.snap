// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Component Test Component renders properly 1`] = `
<header class="v-sheet theme--light v-toolbar v-toolbar--flat v-app-bar v-app-bar--fixed" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px; border-bottom: 4px solid #1976D2;">
  <div class="v-toolbar__content" style="height: 64px;"><button type="button" class="v-app-bar__nav-icon v-btn v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu theme--light"></i></span></button>
    <div class="spacer"></div>
    <div class="v-input v-input--hide-details theme--light v-input--selection-controls v-input--checkbox">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-9" role="checkbox" type="checkbox" value="">
            <div class="v-input--selection-controls__ripple"></div>
          </div>
        </div>
      </div>
    </div><button type="button" class="v-btn v-btn--has-bg theme--light v-size--default" style="height: 100%;"><span class="v-btn__content"><img src="/images/icon/profile.svg"><div class="v-toolbar__title">Dr. Jaylon Parker</div></span></button>
  </div>
</header>
`;
